src/components/tools/hooks/use-measure-text.ts

```diff
@@ -17,3 +17,71 @@ export const useMeasureText = (font = defaultFont) => {
     return measureText(text, font);
   }, [font]);
 };
+
+// Removes leading and trailing whitespace and replaces all other white space with a single space
+const trimWhitespace = (text:string) => {
+  return text.trim().replace(/\s\s+/g, ' ');
+};
+
+// Caches values for measuring lines. First key is font, second is text, third is width, fourth is maxLines.
+const lineCache: Record<string, Record<string, Record<number, Record<number, number>>>> = {};
+
+// Approximates the number of lines a given string will take when rendered in a div of the given width
+export const measureTextLines = (text: string, width: number, font = defaultFont, maxLines = 100) => {
```

Comment:
This is awesome - perhaps it will be something we will leverage in other places.